src/main/java/bio/terra/service/dataset/flight/ingest/IngestUtils.java

```diff
@@ -73,6 +78,71 @@ public static DatasetTable getDatasetTable(FlightContext context, Dataset datase
     return optTable.get();
   }
 
+  /**
+   * We want to validate and sanitize the ingest source file We are looking for a url that matches
+   * the following specification: https://{host}/{container}/{blobName}.csv or
+   * https://{host}/{container}/{blobName}.json NOTE: this definition is probably going to be too
+   * strict. There are valid blob urls that do not match this specification and we may be able to
+   * expand this definition with more testing/requirement gathering
+   *
+   * @param URL
+   * @return
+   */
+  public static BlobUrlParts validateBlobAzureBlobFileURL(String URL) {
+    BlobUrlParts blobUrlParts;
+    try {
+      blobUrlParts = BlobUrlParts.parse(URL);
+    } catch (IllegalArgumentException ex) {
+      throw new InvalidBlobURLException("Blob URL parse failed due to malformed URL.", ex);
+    }
+    String scheme = "https";
+    if (!StringUtils.equals(blobUrlParts.getScheme(), scheme)) {
+      throw new InvalidBlobURLException("Ingest source is not a valid blob url: '" + URL + "'");
```

Comment:
Mentioning that the URL requires https in the error message would be helpful here

Reply:
I was trying to err on the side of not providing too much info about the validation to the user. (Security course strikes again). But, I think I could safely provide a bit more information.

Reply:
I suppose since our repo is open source, cryptic error messages do not help as much.